File: Easy/1758. Minimum Changes To Make Alternating Binary String.java
class Solution {
    public int minOperations(String s) {
        final char[] sToChars = s.toCharArray();
        int startWithZero = 0;
        int startWithOne = 0;
        int n = sToChars.length;
        for (int i = 0; i < n; i++) {
            if (i % 2 != 0) {
                if (sToChars[i] == '0') {
                    startWithZero += 1;
                } else {
                    startWithOne += 1;
                }
            } else {
                if (sToChars[i] == '0') {
                    startWithOne += 1;
                } else {
                    startWithZero += 1;
                }
            }
        }
        return Math.min(startWithZero, startWithOne);
    }
}

File: Easy/1758. Minimum Changes To Make Alternating Binary String.py
class Solution:
    def minOperations(self, s: str) -> int:
        count = 0
        for i, c in enumerate(s):
            if i % 2:
                if c == '1':
                    count += 1
            else:
                if c == '0':
                    count += 1
        return min(count, len(s) - count)

class Solution1:
    def minOperations(self, s: str) -> int:
        start_with_zero = 0
        start_with_one = 0
        for i, c in enumerate(s):
            if i % 2:
                if c == '1':
                    start_with_zero += 1
                else:
                    start_with_one += 1
            else:
                if c == '0':
                    start_with_zero += 1
                else:
                    start_with_one += 1
        return min(start_with_zero, start_with_one)